File: projet-service/src/main/java/org/sid/projetservice/service/CentreService.java
package org.sid.projetservice.service;

import org.sid.projetservice.model.Centre;
import reactor.core.publisher.Flux;
import reactor.core.publisher.Mono;

public interface CentreService {

    Flux<Centre> getAll();

    Mono<Centre> getCentreById(String idC);

    Mono<Centre> saveCentre(Centre centre);

    Mono<Centre> findCentreById(String id);

    Mono<Long> countAllCentres();

    Flux<Centre> getCentresByPage(Integer page, Integer size);
}

File: front-end_WCP/src/app/core/service/centre.service.ts
import {Injectable} from '@angular/core';
import {HttpClient} from '@angular/common/http';
import {CENTRES_URLS, PROJETS_URLS} from './api.url.config';
import {CentreModel} from '../model/centre.model';
import {Observable} from 'rxjs';

@Injectable()
export class CentreService {
  constructor(private http: HttpClient) {}

  saveCentre(centre: CentreModel): Promise<any> {
    return this.http.post(CENTRES_URLS.ADD_CENTRE_URL, centre, {responseType: 'text'})
      .toPromise()
      .then((response) => {
        return response;
      });
  }

  updateCentre(centre: CentreModel): Promise<any> {
    return this.http.put(CENTRES_URLS.UPDATE_CENTRE_URL, centre, {responseType: 'text'})
      .toPromise()
      .then((response) => {
        return response;
      });
  }

  getCentreById(idC: string): Observable<any> {
    return this.http.get(CENTRES_URLS.GET_CENTRE_ID_URL + '/' + idC);
  }

  countAllCentres(): Observable<any> {
    return this.http.get(CENTRES_URLS.COUNT_CENTRES_URL);
  }

  getAllCentres(page: number, size: number): Observable<any> {
    return this.http.get(CENTRES_URLS.GET_CENTRES_URL + '?page=' + page + '&size=' + size);
  }
}
